### PR TITLE
Improve luna property handling

### PR DESCRIPTION
**Pull request by eric-anderson**
* Include all properties instead of the ones just under properties.entity Keys are named with the full path, so it's easy to tell which properties are entity properties.

* Handle properties with varying types, for int/float it takes the most general type. For other cases it takes the first and logs a message since there are likely to be problems.

### CODE REVIEW

File: lib/sycamore/sycamore/query/schema.py [hunk 1]
```diff
@@ -31,32 +35,55 @@ def get_schema(self) -> OpenSearchSchema:
         schema = self._client.get_field_mapping(fields=["*"], index=self._index)
         query = OpenSearchQuery()
 
+        logger.debug(f"Getting schema for index {self._index}")
         # Fetch example values.
         query["index"] = self._index
         query["query"] = {"query": {"match_all": {}}, "size": self.NUM_EXAMPLES}
         random_sample = self._query_executor.query(query)["result"]["hits"]["hits"]
+        # logger.debug(f"Random sample came back with {random_sample}")
```
Comment (baitsguy) [inline lib/sycamore/sycamore/query/schema.py:43]: delete?

File: lib/sycamore/sycamore/utils/nested.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+            return None
+        if keys[0] not in d:
+            return None
+        d = d[keys[0]]
+        keys = keys[1:]
+
+    return d
+
+
+def dotted_lookup(d: Any, keys: str) -> Any:
```
Comment (baitsguy) [inline lib/sycamore/sycamore/utils/nested.py:17]: can you consolidate with https://github.com/aryn-ai/sycamore/blob/23747b1854523d38c785950f4ac9ec78307eec4b/lib/sycamore/sycamore/data/document.py#L217 ? 2 methods doing similar stuff

File: lib/sycamore/sycamore/tests/unit/query/test_schema.py [hunk 3]
```diff
@@ -73,6 +96,16 @@ def test_opensearch_schema():
         }
     }
 
+    # Verify we can handle schems where int and float are both used
```
Comment (mdwelsh) [inline lib/sycamore/sycamore/tests/unit/query/test_schema.py:99]:
```suggestion
    # Verify we can handle schemas where int and float are both used
```

File: lib/sycamore/sycamore/utils/nested.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+from typing import Any
```
Comment (mdwelsh) [inline lib/sycamore/sycamore/utils/nested.py:1]: Seems odd that this would need to be its own file.

**Review by mdwelsh — APPROVED** [review]

File: lib/sycamore/sycamore/query/schema.py (hunk 1 above)
Comment (eric-anderson) [reply lib/sycamore/sycamore/query/schema.py:43]: Done.

File: lib/sycamore/sycamore/utils/nested.py (hunk 2 above)
Comment (eric-anderson) [reply lib/sycamore/sycamore/utils/nested.py:17]: Done. I thought I had a todo to do that. Turned out to be not too hard.

File: lib/sycamore/sycamore/tests/unit/query/test_schema.py (hunk 3 above)
Comment (eric-anderson) [reply lib/sycamore/sycamore/tests/unit/query/test_schema.py:99]: Done.

File: lib/sycamore/sycamore/utils/nested.py (hunk 4 above)
Comment (eric-anderson) [reply lib/sycamore/sycamore/utils/nested.py:1]: I agree, but utils seems like the right place and none of the files in there looked like a good choice.

**Pull request merged**